docs: clarify async instrumentation register request (#90039)
vercel/next.js: docs/01-app/03-api-reference/03-file-conventions/instrumentation.mdx

### `register` (optional)

> The file exports a `register` function that is called **once** during server startup and must complete before the server is ready to handle requests. `register` can be an async function.

Reviewer: ```suggestion
The file exports a `register` function that is called **once** when a new Next.js server instance is initiated, and must complete before the server is ready to handle requests. `register` can be an async function.
```

> The file exports a `register` function that is called **once** when a new Next.js server instance is initiated. `register` can be an async function, and Next.js will wait for it to finish before handling incoming requests.

Reviewer: ```suggestion
The file exports a `register` function that is called **once** during server startup and must complete before the server is ready to handle requests. `register` can be an async function.
```
